packages/php/email-editor/tests/integration/Engine/Personalizer_Test.php

```
@@ -433,4 +433,213 @@ public function testParsingPersonalizationTagAttributes(): void {
+	 * where $ and \ are special (backreferences like $1 or \1); an unescaped "$10"
+	 * would silently resolve to an empty backreference. Only the $ case is asserted
+	 * end-to-end: a backslash cannot be observed in the output either way, because
+	 * esc_url() strips it from href attributes.
+	 */
+	public function testDataLinkHrefValueWithRegexSpecialCharsIsInsertedLiterally(): void {
+		$this->tags_registry->register(
+			new Personalization_Tag(
+				'Deal URL',
+				'woocommerce/deal-url',
+				'Store',
+				function ( $context, $args ) { // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter.FoundAfterLastUsed -- Parameters unused in this test.
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$context'. (undefined)

(UnusedFormalParameter)